R/dashboard.Rmd

```diff
@@ -1,453 +1,516 @@
-
-Column {data-width=200 .sidebar}
-----------------------------------------
-
-```{r}
-comm = dbSendQuery(mydb, "SELECT Principle_Exports FROM tarif_r GROUP BY Principle_Exports")
-pExport <- fetch(comm, n=-1)
-
-selectInput(inputId = "selectedPrincipleExport", label = "Select Principle Export", choices = pExport$Principle_Exports)
-
-```
-
-
-
-
-
-Import Trends by Goods
-========================================
-
-
-Row
-----------------------------------------
-
-Column {data-width=200 .sidebar}
-----------------------------------------
-
-
-Download dataset 
-========================================
-
-
-Row
-----------------------------------------
-
-Column {data-width=200 .sidebar}
-----------------------------------------
-
-
-
-
+---
+title: "Trade Statistics"
+output: 
+  flexdashboard::flex_dashboard:
+    orientation: rows
+    vertical_layout: scroll
+    social: [ "twitter", "facebook", "menu"]
+    source_code: embed
+    runtime: shiny
+    favicon: favicon.ico
+    
+params:
+    useSQL: FALSE # Select whether to load data from SQL server
+    startDate: "2019-01-01" # Start date for latest month (YYYY-dd-mm)
+    endDate: "2019-01-31" # End date for latest month (YYYY-dd-mm)
+---
+
+```{r setup, include=FALSE}
+# Load libraries
+library(flexdashboard) # Building the dashboard layout (useful to add comment for why we need each package here)
+library(knitr) # Rendering Rmarkdown
+library(DT) # Creating html formatted tables
+library(rpivotTable)
+library(ggplot2) # Plotting data
+library(plotly) # Plotting data interactively
+library(dplyr) # Processing data
+library(openintro)
+library(highcharter)
+library(ggvis)
+library(gganimate)
+library(shiny) # Adding interactive elements to dashboard
+library(RMySQL) # Interacting with local SAL database
+library(lubridate) # Working with dates
+library(openxlsx) # Read excel formatted files
+
+# Note repository location
+repository <- dirname(dirname(rstudioapi::getSourceEditorContext()$path))
+
+# Define color palette
+myColors <- c("blue", "#FFC125", "darkgreen", "darkorange")
+
+# Convert the latest month dates to dates
+params$startDate <- as.Date(params$startDate)
+params$endDate <- as.Date(params$endDate)
+```
+
+
+```{r load data and connect to SQL database}
+
+# Load data
+trades <- read.csv(file.path(repository, "data", "secure", "trades.csv")) # Note that csv files need more informative names
+topten <- read.csv(file.path(repository, "data", "secure", "tab.csv")) # Variables need more informative names
+imp <- read.csv(file.path("data", "secure", "country_year_import.csv")) # I don't have access to this file
+exp <- read.csv(file.path("data", "secure", "country_year_export.csv")) # I don't have access to this file
+
+# Connect to local SQL database if requested
+mydb <- NULL
+historicImports <- NULL
+historicExports <- NULL
+historicImportsLatestMonth <- NULL
+historicExportsLatestMonth <- NULL
+principleCOmmodities <- NULL
+if(params$useSQL){

```

Comment:
Using the `params$useSQL` in code chunk to load data either from SQL or from local files